### PR TITLE
ansible: Add async with timeout for OVA related tasks

### PR DESCRIPTION
**Pull request by almusil**
OVA is invoking tasks that are running for a long time
as script or command, this can result in ssh closing connection
before the task has a chance to finish. Use async on those
commands which makes ansible to poll every 15 seconds to check
if the command succeeded.

Bug-Url: https://bugzilla.redhat.com/2056052
Signed-off-by: Ales Musil <amusil@redhat.com>

### CODE REVIEW

**Review by mwperina — APPROVED** [review]
+1

**Review by ahadas — APPROVED** [review]
this looks good but it really looks familiar - @liranr23 , there was something similar that we added to the export/import ova flows, no?

**Comment by almusil** [comment]
Hmm async does not work for `script`, hmm I wonder if we could use `command` instead

**Review by mwperina — APPROVED** [review]
+1

**Comment by almusil** [comment]
Oh I had a wrong understanding how the `script` works, it takes something that is present locally and executes it on remote host. Unfortunate thing is that this does not support `async` and switch for `command` won't work either because command runs something that is present on the remote host. We would need to copy the file first then execute and remove it in the end.

File: packaging/ansible-runner-service-project/project/roles/ovirt-ova-extract/tasks/main.yml [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
 - name: Check OVA extraction process result
   fail:
     msg: "Failed to extract OVA file"
   when: extraction_result.rc is defined and extraction_result.rc != 0
+
+always:
+  - name: Remove temp directory
+    file:
+      state: absent
+      path: "{{ ova_temp.path }}"
```
Comment (mnecas) [inline packaging/ansible-runner-service-project/project/roles/ovirt-ova-extract/tasks/main.yml:33]: To use `always` you need to put previous code to the `block`.
https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html#id3 
This goes for all changed files.

File: packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+- name: Create temporary directory
+  tempfile:
+    state: directory
+    suffix: ova
+  register: ova_temp
+
+- name: Copy pack_ova.py to temp directory
+  copy:
+    src: pack_ova.py
+    dest: "{{ ova_temp.path }}/pack_ova.py.py"
```
Comment (mnecas) [inline packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml:11]: Watch out for the double `.py` please change it to `pack_ova.py`

**Review by mnecas — CHANGES_REQUESTED** [review]

File: packaging/ansible-runner-service-project/project/roles/ovirt-ova-extract/tasks/main.yml (hunk 1 above)
Comment (almusil) [reply packaging/ansible-runner-service-project/project/roles/ovirt-ova-extract/tasks/main.yml:33]: Done, thanks

File: packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml (hunk 2 above)
Comment (almusil) [reply packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml]: Ah good catch, thanks

**Comment by liranr23** [comment]
> this looks good but it really looks familiar - @liranr23 , there was something similar that we added to the export/import ova flows, no?

No, we moved it in the engine to run async (the whole ansible command). But we didn't deal with SSH timeout. The "long" operations are usually the script execution (pack.ova and so on).

File: packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
+  - name: Create temporary directory
+    tempfile:
+      state: directory
+      suffix: ova
+    register: ova_temp
+
+  - name: Copy pack_ova.py to temp directory
+    copy:
+      src: pack_ova.py
+      dest: "{{ ova_temp.path }}/pack_ova.py"
```
Comment (liranr23) [inline packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml:12]: why we need the temporary directory all the sudden on these flows?

Comment (almusil) [reply packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml:12]: Explained in the comments earlier, but in short `script` does not support `async`. In order to run `command` we need to copy those scripts to host.

Comment (liranr23) [reply packaging/ansible-runner-service-project/project/roles/ovirt-ova-pack/tasks/main.yml:12]: Right right, thanks. I missed it.

**Review by liranr23 — APPROVED** [review]

**Comment by liranr23** [comment]
By the way, will [changing the ssh_connection in ansible](https://docs.ansible.com/ansible/2.4/intro_configuration.html#openssh-specific-settings) will resolve it? [Referencing HE-deployment](https://bugzilla.redhat.com/show_bug.cgi?id=1678693) . Maybe something in the configuration will solve it?

**Comment by almusil** [comment]
> By the way, will [changing the ssh_connection in ansible](https://docs.ansible.com/ansible/2.4/intro_configuration.html#openssh-specific-settings) will resolve it? [Referencing HE-deployment](https://bugzilla.redhat.com/show_bug.cgi?id=1678693) . Maybe something in the configuration will solve it?

There is already `ServerAliveInterval=240` in [config](https://github.com/oVirt/ovirt-engine/blob/master/packaging/ansible-runner-service-project/project/ansible.cfg) and that didn't help , I was not able to find any other ssh option related to keep alive.

**Pull request merged**